src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -458,6 +458,14 @@ protected function instantiateObject(array &$data, string $class, array &$contex
 
         unset($context['has_constructor']);
 
+        if (!$reflectionClass->isInstantiable()) {
```

Comment:
What about putting it at the top of the method to fail as fast as possible?

Reply:
Can't add it to the top because in case of Static classes where the constructor is private for instance the `StaticConstructorDummy`.
Would have to also check not static so not sure it makes sense but happy to do so if you think it's worth it

Reply:
Indeed, nice catch!